test/Riok.Mapperly.Tests/_snapshots/ReferenceHandlingTest.CustomHandlerWithObjectFactoryShouldWork#Mapper.g.verified.cs

```diff
@@ -7,7 +7,7 @@ public partial class Mapper
     {
         if (refHandler.TryGetReference<global::A, global::B>(a, out var existingTargetReference))
             return existingTargetReference;
-        var target = CreateB();
+        var target = new global::B();
```

Comment:
IMO this shouldn't change. That's also why a lot of unit tests fail.
The problem is that when the context (`MappingBuilderContext`) for a user mapping is created the object factories are still empty and not yet extracted. This could probably be fixed by making `ObjectFactoryCollection` mutable and removing `ObjectFactoryCollection.Empty` as well as the constructor with the collection and instead adding a public method to add entries. I'm not super happy making the object factory collection mutable but can't currently think of a better solution. Do you have a better idea?

Reply:
Thanks for spotting. My eyes saw a different change `new B -> new global::B`

Reply:
I solved it by dropping global `_objectFactories`. My attempt to turn it to mutable affected the test state and some tests started to fail randomly.

![image](https://github.com/riok/mapperly/assets/3594540/836553f2-f420-419d-91e6-0b97a589de78)

Reply:
I really like this approach and it can even be simplified further (instead of returning / passing the user mappings `_mappings.UserMappings` could be used).